soroban-sdk/src/auth.rs

```diff
@@ -13,3 +15,21 @@ impl Context {
         Address::from_contract_id(&self.contract)
     }
 }
+
+/// Auth interface that a contract implements to support being used for custom
+/// auth.
+///
+/// Once a contract implements the interface, the address of the contract can be
+/// used for addresses that have [`Address::require_auth`] called.
+pub trait Interface<E>
```

Comment:
Unlike `Context`, I feel like this could benefit from a more specific name (like `CustomAccountInterface` or something). `auth::Interface` doesn't make much sense to me.

Reply:
👍🏻 Changed.